### PR TITLE
New: `no-extra-label` rule (fixes #5059)

### LINKED ISSUE #5059: Rule Proposal: `no-extra-label`
From #5052.

This rule warns labels of `break` and `continue` if we can remove the label merely.

The following patterns are considered problems:

```js
/*eslint no-extra-label: 2*/

A:
for (let i = 0; i < 10; ++i) {
    if (foo()) {
        break A;  /*error The label 'A' is unnecessary.*/
    }
    bar();
}

B:
switch (a) {
    case 0:
        foo();
        break B;  /*error The label 'B' is unnecessary.*/
}
```

The following patterns are not considered problems:

```js
/*eslint no-extra-label: 2*/

A:
for (let i = 0; i < 10; ++i) {
    for (let j = 0; j < 10; ++j) {
        if (foo()) {
            break A;
        }
        bar();
    }
}

B:
switch (a) {
    case 0:
        for (let j = 0; j < 10; ++j) {
            if (foo()) {
                break B;
            }
        }
        break;
}

C: {
    if (foo()) {
        break C;
    }
    bar();
}
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30172818-rule-proposal-no-extra-label?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mysticatea**
fixes #5059.

### CODE REVIEW

File: docs/rules/no-extra-label.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Disallow Unnecessary Labels (no-extra-label)
+
+We don't need to use labels in order to control flow of the direct containing loop or switch statement.
```
Comment (nzakas) [inline docs/rules/no-extra-label.md]: Let's try to clean this up a little, something like:

> When a loop is directly labeled, the label becomes unnecessary. You can achieve the same result by removing the label and using `break` or `continue` without a label.

**Comment by nzakas** [comment]
LGTM. Just want to clean up the docs a bit.

File: docs/rules/no-extra-label.md (hunk 1 above)
Comment (platinumazure) [inline docs/rules/no-extra-label.md]: "directly labeled"? Not sure that's much clearer.

Maybe it should say, "If a loop contains no nested loops, labeling the loop is unnecessary" or similar?

Comment (nzakas) [inline docs/rules/no-extra-label.md]: Fine by me.

**Comment by mysticatea** [comment]
I updated it.

**Pull request merged**